File: library/lib_module_common/src/main/java/org/cchao/common/IApplication.java
package org.cchao.common;

import android.app.Application;
import android.content.Context;

import java.lang.reflect.Method;

/**
 * @author cchen6
 * @Date on 2019/8/2
 * @Description
 */
public class IApplication extends Application {

    private static IApplication instance;

    private boolean debug = true;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        initStetho();
    }

    private void initStetho() {
        if (!debug) {
            return;
        }
        try {
            Class cls = Class.forName("com.facebook.stetho.Stetho");
            Method method = cls.getDeclaredMethod("initializeWithDefaults", Context.class);
            method.setAccessible(true);
            method.invoke(null, this);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static IApplication getInstance() {
        return instance;
    }

    public void setDebug(boolean debug) {
        this.debug = debug;
    }

    public boolean isDebug() {
        return debug;
    }
}

File: library/lib_module_common/src/main/java/org/cchao/common/IApplication.kt
package org.cchao.common

import android.app.Application
import android.content.Context

/**
 * @author cchen6
 * @Date on 2019/8/2
 * @Description
 */
open class IApplication : Application() {

    var isDebug = true

    companion object {

        lateinit var instance: IApplication
            private set
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        initStetho()
    }

    private fun initStetho() {
        if (!isDebug) {
            return
        }
        try {
            val cls = Class.forName("com.facebook.stetho.Stetho")
            val method = cls.getDeclaredMethod("initializeWithDefaults", Context::class.java)
            method.isAccessible = true
            method.invoke(null, this)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
